src/new-client/src/app/site/[site]/users/manage/page.tsx

```diff
@@ -18,6 +18,11 @@ const AssignRolesPage = async ({ params, searchParams }: UserPageProps) => {
   const site = await fetchSite(params.site);
   const siteMoniker = site?.name ?? `Site ${params.site}`;
 
+  const permissions = await fetchPermissions(params.site);
+  if (!permissions.includes('users:manage')) {
+    throw new Error('Forbidden: You lack the necessary permissions');
+  }
```

Comment:
We might be able to create some repeatable pattern for gating entire pages behind permissions - I think that might be a tad out of scope for this PR though?